forum/qa-include/pages/message.php

```diff
@@ -1,257 +1,258 @@
             $userLevel = qa_get_logged_in_level();
             if($userLevel < QA_USER_LEVEL_EDITOR) {
                 return include QA_INCLUDE_DIR.'qa-page-not-found.php';
-            }	
+            }    
 
         }
 
 
-//	Check that we have permission and haven't reached the limit, but don't quit just yet
+//    Check that we have permission and haven't reached the limit, but don't quit just yet
 
-	switch (qa_user_permit_error(null, QA_LIMIT_MESSAGES)) {
-		case 'limit':
-			$pageerror = qa_lang_html('misc/message_limit');
-			break;
+    switch (qa_user_permit_error(null, QA_LIMIT_MESSAGES)) {
+        case 'limit':
+            $pageerror = qa_lang_html('misc/message_limit');
+            break;
 
-		case false:
-			break;
+        case false:
+            break;
 
-		default:
-			$pageerror = qa_lang_html('users/no_permission');
-			break;
-	}
-
-
-//	Process sending a message to user
-
-	// check for messages or errors
-	$state = qa_get_state();
-	$messagesent = $state == 'message-sent';
-	if ($state == 'email-error')
-		$pageerror = qa_lang_html('main/email_error');
-
-	if (qa_post_text('domessage')) {
-		$inmessage = qa_post_text('message');
-
-		if (isset($pageerror)) {
-			// not permitted to post, so quit here
-			$qa_content['error'] = $pageerror;
-			return $qa_content;
-		}
-
-		if ( !qa_check_form_security_code('message-'.$handle, qa_post_text('code')) )
-			$pageerror = qa_lang_html('misc/form_security_again');
-
-		else {
-			if (empty($inmessage))
-				$errors['message'] = qa_lang('misc/message_empty');
-
-			if (empty($errors)) {
-				require_once QA_INCLUDE_DIR.'db/messages.php';
-				require_once QA_INCLUDE_DIR.'app/emails.php';
-
-				if (qa_opt('show_message_history'))
-					$messageid = qa_db_message_create($loginuserid, $toaccount['userid'], $inmessage, '', false);
-				else
-					$messageid = null;
+        default:
+            $pageerror = qa_lang_html('users/no_permission');
+            break;
+    }
 
-				$canreply = !(qa_get_logged_in_flags() & QA_USER_FLAGS_NO_MESSAGES);
-				if ($toaccount['pwemail'] == 1) {
-					$more = strtr(qa_lang($canreply ? 'emails/private_message_reply' : 'emails/private_message_info'), array(
-						'^f_handle' => $fromhandle,
-						'^url' => qa_path_absolute($canreply ? ('message/'.$fromhandle) : ('user/'.$fromhandle)),
-					));
 
-					$subs = array(
-						'^message' => $inmessage,
-						'^f_handle' => $fromhandle,
-						'^f_url' => qa_path_absolute('user/'.$fromhandle),
-						'^more' => $more,
-						'^a_url' => qa_path_absolute('account'),
-					);
+//    Process sending a message to user
 
-					qa_send_notification(
-						$toaccount['userid'],
-						$toaccount['email'],
-						$toaccount['handle'],
-						qa_lang('emails/private_message_subject'),
-						qa_lang('emails/private_message_body'),
-						$subs
-					);
-				}
-				$messagesent = true;
+    // check for messages or errors
+    $state = qa_get_state();
+    $messagesent = $state == 'message-sent';

```

Comment:
powinno się stosować a) porównywanie strict (===) b) yoda condition

Reply:
> b) yoda condition

Czy mógłbyś podać jakiś standard lub zalecenia, które faktycznie zalecają taki styl pisania warunków? Osobiście udało mi się znaleźć jedynie forka PSR (ach, te standardy, zawsze podwójne…), który wręcz odradza używania → https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#comparison

Reply:
Tak z ciekawości: dlaczego "powinno" się stosować yoda condition? Jakiś standard to określa? Bo nie wydaje mi się abyśmy coś takiego przyjęli i później wychodzą niejasności bo każdy robi inaczej.

Reply:
@Comandeer nie jest to standard określony w, o ile wiem, żadnym PSR. Jednakże w każdej firmie w której byłem był to standard wewnętrzny. Jasne, można powiedzieć, że to jest wydziwianie. Jednak z czasem uważam używanie Yoda conditions za coś ważnego, coś co jest bardziej działaniem prewencyjnym niż jakościowym.
W pewien sposób ta problematyka jest tu omówiona: https://knowthecode.io/yoda-conditions-yoda-not-yoda

Odnośnie "additions" to jak ze wszystkim, z "human readable" mógłbym dyskutować - najpewniej nie tylko ja. To problem pokroju taby czy spacje. 
Na podstawie samego niezachowania yoda conditions nie odrzuciłbym MR - to tylko komentarz, a MR został odrzucony przez inne rzeczy, takie jak np stosowanie gwiazdek w sqlkach.

Reply:
> Jednakże w każdej firmie w której byłem był to standard wewnętrzny.

A u mnie zawsze jako standard wewnętrzny jest zakaz yoda conditions ;)

Reply:
Dlatego mówię - to temat pokroju "taby czy spacje". Sugeruję to, co uznaję za dobre. Myślę, że to jest mało istotne. Nie ma sensu wszczynać większych dyskusji - po prostu mechanicznie zauważam brak yoda conditions w trakcie code review, o czym informuję - tak robię w pracy, tak też robię tu ;)